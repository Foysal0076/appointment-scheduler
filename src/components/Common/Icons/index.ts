export { SearchIcon } from './SearchIcon'
export { ToggleDarkIcon } from './ToggleDarkIcon'
export { ToggleLightIcon } from './ToggleLightIcon'
