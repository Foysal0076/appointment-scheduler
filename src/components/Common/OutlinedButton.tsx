import { cva } from 'class-variance-authority'
import clsx from 'clsx'

import Spinner from '@/components/Common/Spinner'

type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
  color?: 'primary' | 'secondary' | 'tertiary' | 'danger' | 'success'
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl'
  fullWidth?: boolean
  loading?: boolean
}

const ButtonVariants = cva(
  'rounded bg-none border-2 focus:ring-4 disabled:active:transform-none disabled:cursor-not-allowed dark:hover:bg-neutral-20/50 transition-all duration-300 text-nowrap focus:outline-none font-medium active:scale-[.9]',
  {
    variants: {
      color: {
        primary:
          'hover:bg-primary-50 focus:ring-primary-100 border-primary-500/80 text-primary-500  focus:ring-primary-100 dark:text-primary-200 dark:border-primary-200 disabled:text-primary-200 disabled:border-primary-100 disabled:hover:bg-transparent dark:disabled:bg-primary-50/20 dark:disabled:border-primary-50/10 dark:disabled:text-primary-100/60',
        secondary:
          'hover:bg-secondary-50 focus:ring-secondary-100 border-secondary-500/80 text-secondary-500  focus:ring-secondary-100 dark:text-secondary-200 dark:border-secondary-200 disabled:text-secondary-200 disabled:border-secondary-100 disabled:hover:bg-transparent dark:disabled:bg-secondary-50/20 dark:disabled:border-secondary-50/10 dark:disabled:text-secondary-100/60',
        tertiary:
          'hover:bg-tertiary-50 focus:ring-tertiary-100 border-tertiary-500/80 text-tertiary-500  focus:ring-tertiary-100 dark:text-tertiary-200 dark:border-tertiary-200 disabled:text-tertiary-200 disabled:border-tertiary-100 disabled:hover:bg-transparent dark:disabled:bg-tertiary-50/20 dark:disabled:border-tertiary-50/10 dark:disabled:text-tertiary-100/60',
        danger:
          'hover:bg-danger-50 focus:ring-danger-100 border-danger-500/80 text-danger-500  focus:ring-danger-100 dark:text-danger-200 dark:border-danger-200 disabled:text-danger-200 disabled:border-danger-100 disabled:hover:bg-transparent dark:disabled:bg-danger-50/20 dark:disabled:border-danger-50/10 dark:disabled:text-danger-100/60',
        success:
          'hover:bg-success-50 focus:ring-success-100 border-success-500/80 text-success-500  focus:ring-success-100 dark:text-success-200 dark:border-success-200 disabled:text-success-200 disabled:border-success-100 disabled:hover:bg-transparent dark:disabled:bg-success-50/20 dark:disabled:border-success-50/10 dark:disabled:text-success-100/60',
      },
      size: {
        xs: 'h-[1.75rem] text-xs px-4',
        sm: 'h-[2rem] text-sm px-4',
        md: 'h-[2.5rem] text-md px-4',
        lg: 'h-[3.125rem] text-lg px-4',
        xl: 'h-[3.5rem] text-lg px-4',
      },
    },
    defaultVariants: {
      color: 'primary',
      size: 'md',
    },
  }
)

const OutlinedButton = ({
  color = 'primary',
  size = 'md',
  loading = false,
  className = '',
  children,
  ...buttonProps
}: ButtonProps) => {
  return (
    <button
      className={clsx(
        `relative select-none`,
        `${ButtonVariants({ color, size })}`,
        `${className}`
      )}
      disabled={loading}
      {...buttonProps}>
      {loading && (
        <div className='absolute inset-0 flex items-center justify-center'>
          <Spinner className='text-current' />
        </div>
      )}
      <div
        className={clsx(`visible font-Rubik`, {
          invisible: loading,
          'opacity-0': loading,
        })}>
        {children}
      </div>
    </button>
  )
}

export default OutlinedButton
