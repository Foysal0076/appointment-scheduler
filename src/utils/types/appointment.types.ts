import {
  AppointmentFilters,
  AppointmentStatus,
} from '../constants/appointment.constants'

export type AppointmentStatusType =
  (typeof AppointmentStatus)[keyof typeof AppointmentStatus]

export type AppointmentFiltersType =
  (typeof AppointmentFilters)[keyof typeof AppointmentFilters]
