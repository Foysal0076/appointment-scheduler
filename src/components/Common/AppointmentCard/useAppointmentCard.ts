import { useState } from 'react'
import { toast } from 'react-toastify'

import { PutAppointmentBody } from '@/redux/apiQueries/apiQueries.type'
import {
  useDeleteAppointmentMutation,
  useUpdateAppointmentMutation,
} from '@/redux/apiQueries/appointmentQueries'
import { AppointmentStatusType } from '@/utils/types/appointment.types'

export const useAppointmentCard = (
  appointmentId: string,
  startTime: number
) => {
  const [updateAppointment, { isLoading: isUpdatingAppointment }] =
    useUpdateAppointmentMutation()

  const [deleteAppointment, { isLoading: isDeletingAppointment }] =
    useDeleteAppointmentMutation()

  const [loadingStatus, setLoadingStatus] =
    useState<AppointmentStatusType | null>(null)

  const onCancelAppointment = async () => {
    try {
      //check if the meeting is already started
      const currentTime = new Date().getTime()
      if (currentTime > startTime) {
        toast.error('You can not cancel a meeting that is past current time')
        return
      }

      const body: PutAppointmentBody = {
        id: appointmentId,
        status: 'cancelled',
      }
      await updateAppointment(body)
    } catch (error) {
      console.log(error)
      toast.error('Something went wrong')
    }
  }

  const onAcceptOrDeclineAppointment = async (
    status: AppointmentStatusType
  ) => {
    try {
      setLoadingStatus(status)
      const body: PutAppointmentBody = {
        id: appointmentId,
        status,
      }
      await updateAppointment(body)
    } catch (error) {
      console.log(error)
      toast.error('Something went wrong')
    } finally {
      setLoadingStatus(null)
    }
  }

  const onDeleteAppointment = async () => {
    try {
      await deleteAppointment(appointmentId)
    } catch (error) {
      console.log(error)
      toast.error('Something went wrong')
    }
  }

  return {
    loadingStatus,
    isUpdatingAppointment,
    isDeletingAppointment,
    onCancelAppointment,
    onAcceptOrDeclineAppointment,
    onDeleteAppointment,
  }
}
