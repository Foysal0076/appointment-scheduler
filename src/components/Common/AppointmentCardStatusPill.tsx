import clsx from 'clsx'

import { AppointmentStatusType } from '@/utils/types/appointment.types'

type Props = {
  status: AppointmentStatusType
}

const AppointmentCardStatusPill = ({ status }: Props) => {
  return (
    <div
      className={clsx(
        'select-none rounded-full border px-2 py-1 text-sm font-bold capitalize tracking-wide text-white transition-colors duration-200',
        { 'border-success-500 bg-success-400': status === 'approved' },
        { 'border-danger-500 bg-danger-400': status === 'rejected' },
        { 'border-warning-500 bg-warning-400': status === 'cancelled' },
        { 'border-info-500 bg-info-400': status === 'pending' }
      )}>
      {status}
    </div>
  )
}

export default AppointmentCardStatusPill
