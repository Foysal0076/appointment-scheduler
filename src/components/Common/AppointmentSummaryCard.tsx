import Button from '@/components/Common/Button'
import OutlinedButton from '@/components/Common/OutlinedButton'
import { HOUR_FORMAT } from '@/utils/constants/appointment.constants'
import { formatDate, formatTime } from '@/utils/helpers'
import { HourFormat } from '@/utils/types/appointment.types'

type AppointmentSummaryCardProps = {
  title: string
  description: string
  host: string // the person who scheduled the appointment
  guest: string // the person who is invited for the appointment
  audioFile?: File | Blob | null
  startTime: Date
  endTime: Date
  hourFormat?: HourFormat
  onSubmit: () => void
  onCancel: () => void
  loading?: boolean
}

const AppointmentSummaryCard = ({
  host,
  guest,
  audioFile,
  startTime,
  endTime,
  title,
  description,
  hourFormat = HOUR_FORMAT.h12,
  onSubmit,
  onCancel,
  loading,
}: AppointmentSummaryCardProps) => {
  return (
    <div className='flex h-full flex-col justify-between rounded-lg border border-neutral-30 bg-surface-0 p-4 shadow-md dark:bg-surface-100'>
      <div className='grow'>
        <h2 className='h3 mb-2'>{formatDate(startTime)}</h2>
        <p className='h5 mb-4 inline font-bold'>
          {formatTime(startTime, hourFormat)} -{' '}
          {formatTime(endTime, hourFormat)}
        </p>
        <h3 className='h6 mb-4'>{title}</h3>
        <p className='mb-4 line-clamp-4'>{description}</p>
        <div className='mb-4'>
          {audioFile && <audio src={URL.createObjectURL(audioFile)} controls />}
        </div>
        <p className='mb-4 inline text-sm text-neutral-700'>
          Participants: {host} and {guest}
        </p>
      </div>
      <div className='flex justify-end gap-4'>
        <>
          <OutlinedButton color='danger' size='sm' onClick={onCancel}>
            Cancel
          </OutlinedButton>
          <Button size='sm' onClick={onSubmit} loading={loading}>
            Confirm
          </Button>
        </>
      </div>
    </div>
  )
}

export default AppointmentSummaryCard
