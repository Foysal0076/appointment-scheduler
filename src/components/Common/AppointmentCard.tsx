import { useState } from 'react'
import { toast } from 'react-toastify'

import AppointmentCardStatusPill from '@/components/Common/AppointmentCardStatusPill'
import OutlinedButton from '@/components/Common/OutlinedButton'
import { PutAppointmentBody } from '@/redux/apiQueries/apiQueries.type'
import {
  useDeleteAppointmentMutation,
  useUpdateAppointmentMutation,
} from '@/redux/apiQueries/appointmentQueries'
import { formatDate, formatTime } from '@/utils/helpers'
import {
  AppointmentItem,
  AppointmentStatusType,
  HourFormat,
} from '@/utils/types/appointment.types'

type AppointmentCardProps = AppointmentItem & {
  userId: string
  hourFormat?: HourFormat
}

const AppointmentCard = ({
  id,
  title,
  description,
  audioMessage,
  hostInfo,
  guestInfo,
  userId,
  status,
  startTime,
  endTime,
}: AppointmentCardProps) => {
  const isHost = hostInfo.id === userId
  const isRejected = status === 'rejected'
  const isPast = new Date().getTime() > endTime

  const [updateAppointment, { isLoading: isUpdatingAppointment }] =
    useUpdateAppointmentMutation()

  const [deleteAppointment, { isLoading: isDeletingAppointment }] =
    useDeleteAppointmentMutation()

  const [loadingStatus, setLoadingStatus] =
    useState<AppointmentStatusType | null>(null)

  const onCancelAppointment = async () => {
    try {
      //check if the meeting is already started
      const currentTime = new Date().getTime()
      if (currentTime > startTime) {
        toast.error('You can not cancel a meeting that is past current time')
        return
      }
      console.log('cancel appointment')
      const body: PutAppointmentBody = {
        id,
        status: 'cancelled',
      }
      await updateAppointment(body)
    } catch (error) {
      console.log(error)
      toast.error('Something went wrong')
    }
  }

  const onAcceptOrDeclineAppointment = async (
    status: AppointmentStatusType
  ) => {
    try {
      setLoadingStatus(status)
      const body: PutAppointmentBody = {
        id,
        status,
      }
      await updateAppointment(body)
    } catch (error) {
      console.log(error)
      toast.error('Something went wrong')
    } finally {
      setLoadingStatus(null)
    }
  }

  const onDeleteAppointment = async () => {
    try {
      await deleteAppointment(id)
    } catch (error) {
      console.log(error)
      toast.error('Something went wrong')
    }
  }

  return (
    <div className='relative flex h-full flex-col justify-between gap-4 rounded-lg border border-neutral-30 bg-surface-0 p-4 shadow-md dark:bg-surface-100'>
      <div className='absolute right-2 top-2'>
        <AppointmentCardStatusPill status={status} />
      </div>
      <div className='grow'>
        <h2 className='h3 mb-2'>{formatDate(startTime)}</h2>
        <p className='h5 mb-4 font-bold'>
          {formatTime(startTime)} - {formatTime(endTime)}
        </p>
        <h3 className='h6 mb-4'>{title}</h3>
        <p className='mb-4 line-clamp-4 text-base text-neutral-400'>
          {description}
        </p>
        <div className='mb-4'>
          {audioMessage && <audio src={audioMessage} controls />}
        </div>
      </div>
      <div>
        <p className='mb-4 text-xs text-neutral-700 md:text-sm'>
          {isHost && isRejected
            ? `${guestInfo.fullname} rejected this appointment`
            : `Participants: ${hostInfo.fullname}, ${guestInfo.fullname}`}
        </p>
        <div className='flex justify-between gap-4'>
          {isHost ? (
            <>
              <OutlinedButton
                onClick={onDeleteAppointment}
                color='danger'
                size='sm'
                loading={isDeletingAppointment}>
                Delete
              </OutlinedButton>
              <OutlinedButton
                onClick={onCancelAppointment}
                color='tertiary'
                size='sm'
                disabled={status === 'cancelled' || isPast}
                loading={isUpdatingAppointment}>
                {status === 'cancelled' ? 'Cancelled' : 'Cancel'}
              </OutlinedButton>
            </>
          ) : (
            <>
              <OutlinedButton
                onClick={() => onAcceptOrDeclineAppointment('rejected')}
                color='danger'
                size='sm'
                disabled={status === 'rejected' || isPast}
                loading={loadingStatus === 'rejected'}>
                Decline
              </OutlinedButton>
              <OutlinedButton
                onClick={() => onAcceptOrDeclineAppointment('approved')}
                color='success'
                size='sm'
                disabled={status === 'approved' || isPast}
                loading={loadingStatus === 'approved'}>
                {status === 'approved' ? 'Accepted' : 'Accept'}
              </OutlinedButton>
            </>
          )}
        </div>
      </div>
    </div>
  )
}

export default AppointmentCard
