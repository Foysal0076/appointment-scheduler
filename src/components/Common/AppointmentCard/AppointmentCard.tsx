import { useMemo } from 'react'

import { useAppointmentCard } from '@/components/Common/AppointmentCard/useAppointmentCard'
import AppointmentCardStatusPill from '@/components/Common/AppointmentCardStatusPill'
import OutlinedButton from '@/components/Common/OutlinedButton'
import { formatDate, formatTime } from '@/utils/helpers'
import { AppointmentItem, HourFormat } from '@/utils/types/appointment.types'

type AppointmentCardProps = AppointmentItem & {
  userId: string
  hourFormat?: HourFormat
}

export const AppointmentCard = ({
  id,
  title,
  description,
  audioMessage,
  hostInfo,
  guestInfo,
  userId,
  status,
  startTime,
  endTime,
}: AppointmentCardProps) => {
  const isHost = hostInfo.id === userId
  const isRejected = status === 'rejected'
  const isPast = useMemo(() => new Date().getTime() > endTime, [endTime])

  const {
    loadingStatus,
    isUpdatingAppointment,
    isDeletingAppointment,
    onCancelAppointment,
    onAcceptOrDeclineAppointment,
    onDeleteAppointment,
  } = useAppointmentCard(id, startTime)

  return (
    <div className='relative flex h-full flex-col justify-between gap-4 rounded-lg border border-neutral-30 bg-surface-0 p-4 shadow-md dark:bg-surface-100'>
      <div className='absolute right-2 top-2'>
        <AppointmentCardStatusPill status={status} />
      </div>
      <div className='grow'>
        <h2 className='h3 mb-2'>{formatDate(startTime)}</h2>
        <p className='h5 mb-4 font-bold'>
          {formatTime(startTime)} - {formatTime(endTime)}
        </p>
        <h3 className='h6 mb-4'>{title}</h3>
        <p className='mb-4 line-clamp-4 text-base text-neutral-400'>
          {description}
        </p>
        <div className='mb-4'>
          {audioMessage && <audio src={audioMessage} controls />}
        </div>
      </div>
      <div>
        <p className='mb-4 text-xs text-neutral-700 md:text-sm'>
          {isHost && isRejected
            ? `${guestInfo.fullname} rejected this appointment`
            : `Participants: ${hostInfo.fullname}, ${guestInfo.fullname}`}
        </p>
        <div className='flex justify-between gap-4'>
          {isHost ? (
            <>
              <OutlinedButton
                onClick={onDeleteAppointment}
                color='danger'
                size='sm'
                loading={isDeletingAppointment}>
                Delete
              </OutlinedButton>
              <OutlinedButton
                onClick={onCancelAppointment}
                color='tertiary'
                size='sm'
                disabled={status === 'cancelled' || isPast}
                loading={isUpdatingAppointment}>
                {status === 'cancelled' ? 'Cancelled' : 'Cancel'}
              </OutlinedButton>
            </>
          ) : (
            <>
              <OutlinedButton
                onClick={() => onAcceptOrDeclineAppointment('rejected')}
                color='danger'
                size='sm'
                disabled={status === 'rejected' || isPast}
                loading={loadingStatus === 'rejected'}>
                Decline
              </OutlinedButton>
              <OutlinedButton
                onClick={() => onAcceptOrDeclineAppointment('approved')}
                color='success'
                size='sm'
                disabled={status === 'approved' || isPast}
                loading={loadingStatus === 'approved'}>
                {status === 'approved' ? 'Accepted' : 'Accept'}
              </OutlinedButton>
            </>
          )}
        </div>
      </div>
    </div>
  )
}
