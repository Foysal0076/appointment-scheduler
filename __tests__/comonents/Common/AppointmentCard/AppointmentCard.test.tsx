// import 'node-fetch'
import { AppointmentCard } from '@/components/Common/AppointmentCard'
import { store } from '@/redux/store' // Adjust the path to your store
import { AppointmentItem } from '@/utils/types/appointment.types'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'

describe('AppointmentCard component', () => {
  const appointment: AppointmentItem = {
    id: '1',
    title: 'Appointment Title',
    description: 'Appointment Description',
    audioMessage: 'audio.mp3',
    hostInfo: {
      id: 'host1',
      email: 'host1@email.com',
      fullname: 'Host Fullname',
    },
    guestInfo: {
      id: 'guest1',
      email: 'guest1@email.com',
      fullname: 'Guest Fullname',
    },
    hostId: 'host1',
    guestId: 'guest1',
    status: 'approved',
    startTime: 1723066405736,
    endTime: 1723066426790,
  }

  const renderWithProvider = (component: React.ReactNode) => {
    return render(<Provider store={store}>{component}</Provider>)
  }

  test('renders without crashing', () => {
    renderWithProvider(<AppointmentCard userId='userId' {...appointment} />)
  })

  test('displays appointment details correctly', () => {
    const { getByText } = renderWithProvider(
      <AppointmentCard userId='userId' {...appointment} />
    )
    expect(getByText('Appointment Title')).toBeInTheDocument()
    expect(getByText('Appointment Description')).toBeInTheDocument()
    expect(
      getByText('Participants: Host Fullname, Guest Fullname')
    ).toBeInTheDocument()
  })

  test('displays audio message if provided', () => {
    const { getByRole } = renderWithProvider(
      <AppointmentCard userId='userId' {...appointment} />
    )
    expect(getByRole('audio')).toBeInTheDocument()
  })

  test('displays status pill', () => {
    const { getByText } = renderWithProvider(
      <AppointmentCard userId='userId' {...appointment} />
    )
    expect(getByText('approved')).toBeInTheDocument()
  })

  test('displays cancel button for host', () => {
    const { getByText } = renderWithProvider(
      <AppointmentCard userId='host1' {...appointment} />
    )
    expect(getByText('Cancel')).toBeInTheDocument()
  })

  test('displays decline and accept buttons for guest', () => {
    const { getByText } = renderWithProvider(
      <AppointmentCard userId='guest1' {...appointment} status='pending' />
    )
    expect(getByText('Decline')).toBeInTheDocument()
    expect(getByText('Accept')).toBeInTheDocument()
  })

  //snapshot test
  test('matches snapshot', () => {
    const { container } = renderWithProvider(
      <AppointmentCard userId='userId' {...appointment} />
    )
    expect(container).toMatchSnapshot()
  })
})
