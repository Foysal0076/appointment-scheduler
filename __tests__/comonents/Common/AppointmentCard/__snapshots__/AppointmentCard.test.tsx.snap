// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentCard component matches snapshot 1`] = `
<div>
  <div
    class="relative flex h-full flex-col justify-between gap-4 rounded-lg border border-neutral-30 bg-surface-0 p-4 shadow-md dark:bg-surface-100"
  >
    <div
      class="absolute right-2 top-2"
    >
      <div
        class="select-none rounded-full border px-2 py-1 text-sm font-bold capitalize tracking-wide text-white transition-colors duration-200 border-success-500 bg-success-400"
      >
        approved
      </div>
    </div>
    <div
      class="grow"
    >
      <h2
        class="h3 mb-2"
      >
        Aug 8, 2024
      </h2>
      <p
        class="h5 mb-4 font-bold"
      >
        03:33 AM
         - 
        03:33 AM
      </p>
      <h3
        class="h6 mb-4"
      >
        Appointment Title
      </h3>
      <p
        class="mb-4 line-clamp-4 text-base text-neutral-400"
      >
        Appointment Description
      </p>
      <div
        class="mb-4"
      >
        <audio
          controls=""
          role="audio"
          src="audio.mp3"
        />
      </div>
    </div>
    <div>
      <p
        class="mb-4 text-xs text-neutral-700 md:text-sm"
      >
        Participants: Host Fullname, Guest Fullname
      </p>
      <div
        class="flex justify-between gap-4"
      >
        <button
          class="relative select-none rounded bg-none border-2 focus:ring-4 disabled:active:transform-none disabled:cursor-not-allowed dark:hover:bg-neutral-20/50 transition-all duration-300 text-nowrap focus:outline-none font-medium active:scale-[.9] hover:bg-danger-50 focus:ring-danger-100 border-danger-500/80 text-danger-500  focus:ring-danger-100 dark:text-danger-200 dark:border-danger-200 disabled:text-danger-200 disabled:border-danger-100 disabled:hover:bg-transparent dark:disabled:bg-danger-50/20 dark:disabled:border-danger-50/10 dark:disabled:text-danger-100/60 h-[2rem] text-sm px-4"
          disabled=""
        >
          <div
            class="visible font-Rubik"
          >
            Decline
          </div>
        </button>
        <button
          class="relative select-none rounded bg-none border-2 focus:ring-4 disabled:active:transform-none disabled:cursor-not-allowed dark:hover:bg-neutral-20/50 transition-all duration-300 text-nowrap focus:outline-none font-medium active:scale-[.9] hover:bg-success-50 focus:ring-success-100 border-success-500/80 text-success-500  focus:ring-success-100 dark:text-success-200 dark:border-success-200 disabled:text-success-200 disabled:border-success-100 disabled:hover:bg-transparent dark:disabled:bg-success-50/20 dark:disabled:border-success-50/10 dark:disabled:text-success-100/60 h-[2rem] text-sm px-4"
          disabled=""
        >
          <div
            class="visible font-Rubik"
          >
            Accepted
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
